tests/test_http_downloader.py

```diff
@@ -0,0 +1,145 @@
+import tempfile
+from unittest.mock import patch
+
+import pytest
+
+from datasetinsights.io.downloader.http_downloader import HTTPDownloader
+from datasetinsights.io.exceptions import ChecksumError
+
+
+def test_read_checksum_from_local_file():
+    # arrange
+    with tempfile.NamedTemporaryFile(mode="w+") as tmp:
+        tmp.write("123456")
+        tmp.flush()
+        # act
+        checksum = HTTPDownloader.get_checksum_from_file(tmp.name)
+        # assert
+        assert checksum == 123456
+
+
+@pytest.mark.parametrize("filepath", ["http://some/path", "https://some/path"])
+@patch("datasetinsights.io.downloader.http_downloader.download_file")
+@patch.object(HTTPDownloader, "read_checksum_from_txt")
+def test_get_checksum_from_http_source(
+    mock_read_checksum_from_txt, mock_download_file, filepath
+):
+    # arrange
+    mock_read_checksum_from_txt.return_value = 123456
+    # act
+    checksum = HTTPDownloader.get_checksum_from_file(filepath)
+    # assert
+    mock_download_file.assert_called_once()
+    assert checksum == 123456
+
+
+@patch("datasetinsights.io.downloader.http_downloader.download_file")
+@patch.object(HTTPDownloader, "read_checksum_from_txt")
+def test_get_checksum_from_non_existing_file(
+    mock_read_checksum_from_txt, mock_download_file
+):
+    # arrange
+    filepath = "some/wrong/path"
+    # assert
+    with pytest.raises(ValueError):
+        # act
+        HTTPDownloader.get_checksum_from_file(filepath)
+
+    # assert
+    mock_download_file.assert_not_called()
+    mock_read_checksum_from_txt.assert_not_called()
+
+
+@pytest.mark.parametrize(
+    "source_uri",
+    ["http://some/path", "https://some/path", "some/checksum_file.txt"],
```

Comment:
Why do you need a `some/checksum_file.txt` for `source_uri`? Also, the name of this method seems to suggest it tries to run logic without checksum. Why do we need to have both `source_uri=http, https` when the logic of handling either http/https source uri is already covered by different use cases? Ideally, user should be able to provide any random string for `source_uri` and be able to test the behavior when `checksum_file` is not provided.

Reply:
The some/checksum_file.txt can be removed from the test case. The test is about not providing a checksum file and validating that the download and unzip is called properly

Reply:
I don't understand your comment on what is wrong with the test besides some/checksum_file.txt

Reply:
I think @adason mean that we don't need to test for both `http` and `https` since this test doesn't check whether it handles `http` and `https` but it's about whether download and unzip is called properly.

Reply:
Ok in that case we can remove the parametrize method and just have one source_uri